cmd/thanos/rule.go

```diff
@@ -529,11 +538,27 @@ func runRule(
 	// Start UI & metrics HTTP server.
 	{
 		router := route.New()
-		router.Post("/-/reload", func(w http.ResponseWriter, r *http.Request) {
+
+		// redirect from / to /webRoutePrefix
+		if webRoutePrefix != "" {
+			router.Get("/", func(w http.ResponseWriter, r *http.Request) {
+				http.Redirect(w, r, webRoutePrefix, http.StatusFound)
+			})
+		}
+
+		router.WithPrefix(webRoutePrefix).Post("/-/reload", func(w http.ResponseWriter, r *http.Request) {
 			reload <- struct{}{}
 		})
 
-		ui.NewRuleUI(logger, mgr, alertQueryURL.String()).Register(router)
+		flagsMap := map[string]string{
```

Comment:
As this makes sense, I would suggest to remove this from this PR.

First of all it is not relevant to change. Secondly it is kind of wrong as we put here only web flags, not every query flag, so it might be confusing.

Fixing it for both query and ruler is needed but we can do it in next PR to leave this one focused. What do you think? (:

Reply:
Do you suggest removing "grpc-address" and "http-address" values and leaving "web.*" as is?
Or should `flagsMap` be removed? In latter case, i would need to pass "web.*prefix*" flags somehow other than via `flagsMap`.

Reply:
OK, i left the prefix-related flags and added a todo message.

Reply:
sorry, I mean `flagsMap` totally (: we need to fix this, but we can do it later